### PR TITLE
Post-training optimization using OpenVINO added

### PR DESCRIPTION
**Pull request by junwenwu**
Fixes #ISSUE_NUMBER

## Proposed Changes

- 

## Checklist

- [ ] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [ ] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [ ] Non-breaking change (would not break existing functionality): please provide as many details as possible for any breaking change 
- [ ] Function/class source code documentation added/updated
- [ ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [ ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [ ] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [ ] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

File: GANDLF/compute/forward_pass.py
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:77]:
```suggestion
    if "type" in params["model"]:
        if params["model"]["type"].lower() = "torch":
```

File: GANDLF/compute/forward_pass.py
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:86]:
```suggestion
    if params["medcam_enabled"] and params["model"]["type"].lower() = "torch":
```

File: GANDLF/compute/forward_pass.py
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:164]:
```suggestion
                if params["model"]["type"].lower() = "torch":
```

File: GANDLF/compute/forward_pass.py
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:333]:
```suggestion
            if params["medcam_enabled"] and params["model"]["type"].lower() = "torch":
```

File: GANDLF/compute/forward_pass.py
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:389]:
```suggestion
    if params["medcam_enabled"] and params["model"]["type"].lower() = "torch":
```

File: GANDLF/compute/inference_loop.py
Comment (sarthakpati) [inline GANDLF/compute/inference_loop.py:51]:
```suggestion
    if parameters["model"]["type"].lower() = "torch"
```

File: GANDLF/compute/inference_loop.py
Comment (sarthakpati) [inline GANDLF/compute/inference_loop.py:160]:
```suggestion
                    if parameters["model"]["type"].lower() = "torch"
```

File: GANDLF/inference_manager.py
Comment (sarthakpati) [inline GANDLF/inference_manager.py:42]:
```suggestion
        parameters["model"]["type"] = "torch"
```

File: GANDLF/compute/forward_pass.py
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:26]:
```suggestion
    model : if parameters["model"]["type"] == torch, this is a torch.model, otherwise this is OV exec_net
```

**Review by sarthakpati — CHANGES_REQUESTED** [review]
Minor update of `"Torch"` going to `"torch"` for consistency.

File: setup.py
Comment (junwenwu) [reply setup.py:54]: OpenVINO-dev has a dependency of numpy<1.20

Comment (sarthakpati) [reply setup.py:54]: Found the relevant PR, which is a few months old: https://github.com/openvinotoolkit/openvino/pull/8653

It is alluding to some internal case number. Can we get some clarification for this?

**Review by junwenwu — COMMENTED** [review]
Made the suggested changes. Rolled back to earlier version of OpenVINO (2021.3) that does not require lumpy dependency. Will update once the numpy dependency for OpenVINO is updated to 1.21.0

File: setup.py
Comment (dkurt) [inline setup.py:81]: `openvino-dev` installs a lot of dependencies so for ONNX models you can use import from ONNX and install just `openvino` package:

```python
import io
buf = io.BytesIO()

torch.onnx.export(
                model,
                dummy_input.to("cpu"),
                buf,
                opset_version=11,
                export_params=True,
                verbose=True,
                input_names=["input"],
                output_names=["output"],
            )

net = ie.read_network(buf.getvalue(), b"", init_from_buffer=True)
```

Comment (junwenwu) [reply setup.py:81]: @dkurt We used to observe for some .onnx models the performance do not match with OV IR models. Is it still an issue? That's actually the reason we want to use MO and convert it to OpenVINO IR.

Comment (dkurt) [reply setup.py:81]: This is not an issue for latest releases. Can you check for this model?

Comment (junwenwu) [reply setup.py:81]: This gave me the following error that I don't quite understand, since numpy~=1.17.3 shouldn't conflict with the numpy==1.21.0 requirement. 

The conflict is caused by:
[763](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:763)
    gandlf 0.0.14.dev0 depends on numpy==1.21.0
[764](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:764)
    openpatchminer 0.1.6 depends on numpy
[765](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:765)
    openvino-dev 2021.3.0 depends on numpy~=1.17.3
[766](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:766)

[767](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:767)
To fix this you could try to:
[768](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:768)
1. loosen the range of package versions you've specified
[769](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:769)
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
[770](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:770)
2. remove package versions to allow pip attempt to solve the dependency conflict
[771](https://github.com/CBICA/GaNDLF/runs/5276145649?check_suite_focus=true#step:4:771)

Comment (dkurt) [reply setup.py:81]: I mean you can just install OpenVINO independently (`pip install openvino==2021.4.2`) or in different environment to check that `.onnx` and IR performance is different or not.

File: .github/workflows/python-test.yml
Comment (sarthakpati) [inline .github/workflows/python-test.yml:29]: This should probably go before `pip install -e .`...

File: GANDLF/utils/modelio.py
Comment (sarthakpati) [inline GANDLF/utils/modelio.py:62]:
```suggestion
    except RuntimeWarning:
```
To fix the codacy warning.

File: testing/config_segmentation.yaml
Comment (sarthakpati) [inline testing/config_segmentation.yaml:12]: This should be reverted back to `resunet`

**Comment by junwenwu** [comment]
I’m having this error from the test:
ModuleNotFoundError: No module named 'opm'
Do you see this before?

File: GANDLF/compute/forward_pass.py
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:83]:
```suggestion
```
We do not need this, do we?

**Comment by sarthakpati** [comment]
> @sarthakpati Testing is getting the error `FileNotFoundError: File not found: "/home/runner/work/GaNDLF/GaNDLF/testing/data_output/histo_patches/histo_patches_output/1/image/image_patch_2624-2304.png"` for the test case `testing/test_full.py::test_train_segmentation_rad_2d`. Any insights on how should this be fixed?

Hey @junwenwu, I fixed it in: https://github.com/junwenwu/GaNDLF/pull/6

**Review by sarthakpati — DISMISSED** [review]
LGTM, will wait for @Geeks-Sid to take a look before merging.

**Review by Geeks-Sid — APPROVED** [review]

File: GANDLF/compute/forward_pass.py
Comment (junwenwu) [reply GANDLF/compute/forward_pass.py:81]: `params["model"]["type"]` is set to `"torch"` if it is not explicitly set to `"openvino"` in file `GANDLF/utils/parameter_processing.py`

**Comment by sarthakpati** [comment]
Should be good to merge after https://github.com/junwenwu/GaNDLF/pull/8

**Review by Geeks-Sid — APPROVED** [review]

**Pull request merged**
